qa/tasks/rgw_reshard.py

```diff
@@ -0,0 +1,318 @@
+    bucket_id = out['id']
+    num_objects = 0
+    size_kb = 0
+    if (len(out['usage']) > 0):
+        num_objects = out['usage']['rgw.main']['num_objects']
+        size_kb = out['usage']['rgw.main']['size_kb']
+
+    num_shards = get_bucket_num_shards(ctx, client, bucket_name, bucket_id)
+    log.debug("bucket %s id %s num_objects %d size_kb %d num_shards %d", bucket_name, bucket_id, num_objects, size_kb, num_shards)
+    return BucketStats(bucket_name, bucket_id, num_objects, size_kb, num_shards)
+
+
+def  get_user_stats(ctx, client, uid):
+    """
+    function to get user stats
+    """
+    (err, out) = rgwadmin(ctx, client,
+                          ['user', 'stats', '--uid', uid, '--sync-stats'], check_status=True)
+    assert err == 0
+    total_entries = out['stats']['total_entries']
+    total_bytes = out['stats']['total_bytes']
+    total_bytes_rounded = out['stats']['total_bytes_rounded']
+    log.debug("user %s total_entries %d total_bytes %d rounded %d", uid, total_entries, total_bytes, total_bytes_rounded)
+    return UserStats(uid, total_entries, total_bytes, total_bytes_rounded)
+
+def get_bucket_num_shards(ctx, client, bucket_name, bucket_id):
+    """
+    function to get bucket num shards
+    """
+    metadata = 'bucket.instance:' + bucket_name +':' + bucket_id
+    log.debug("metadata %s", metadata)
+    (err, out) = rgwadmin(ctx, client,
+                          ['metadata', 'get', metadata], check_status=True)
+    assert err == 0
+    num_shards = out['data']['bucket_info']['num_shards']
+    log.debug("bucket %s id %s num_shards %d", bucket_name, bucket_id, num_shards )
+    return num_shards
+
+def task(ctx, config):
+    """
+    Test resharding functionality against a running rgw instance.
+    """
+    global log
+
+    log.debug('rgw_reshard config is: %r', config)
+
+    assert ctx.rgw.config, \
+        "rgw_reshard task needs a config passed from the rgw task"
+    config = ctx.rgw.config
+    log.debug('rgw_reshard config is: %r', config)
+
+    clients_from_config = config.keys()
+    log.debug('rgw_reshard config is: %r', config)
+
+    # choose first client as default
+    client = clients_from_config[0]
+    log.debug('rgw_reshard client is: %r', client)
+
+    # once the client is chosen, pull the host name and  assigned port out of
+    # the role_endpoints that were assigned by the rgw task
+    endpoint  = ctx.rgw.role_endpoints[client]
+    remote_host = endpoint.hostname
+    remote_port = endpoint.port
+    log.debug('remote host %r remote port %r', remote_host, remote_port)
+
+    ##
+    user = 'testid'
+    access_key='0555b35654ad1656d804'
+    secret_key='h7GhxuBLTrlhVUyxSPUKUV8r/2EI4ngqJxD7iBdBYLhwluN30JaT3Q=='
+    bucket_name1='myfoo'
+    bucket_name2='mybar'
+    ver_bucket_name ='myver'
+
+    # connect to rgw
+    connection = boto.s3.connection.S3Connection(
+        aws_access_key_id=access_key,
+        aws_secret_access_key=secret_key,
+        is_secure=False,
```

Comment:
if we ever run this in the verify suite (or enable ssl in other suites) we'll need to enable this. in my wip branch for radosgw_admin_rest task, i'm using this line:
```python
        is_secure=True if endpoint.cert else False,
```

Reply:
@cbodley , do you think it should run in verify suite?

Reply:
not really, no - at least not in the same job as s3/swift tests because of the ceph.conf overrides